### PR TITLE
Pick up gateway rc2

### PR DESCRIPTION
**Pull request by howardjohn**
This updates to rc2. all of the changes are around the new HTTP isolation test. This implements new behavior to allow us to pass that case.

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -2143,7 +2143,7 @@ func validateGatewayNames(gatewayNames []string) (errs Validation) {
 			errs = AppendValidation(errs, fmt.Errorf("invalid value for namespace: %q", parts[0]))
 		}
 
-		if !labels.IsDNS1123Label(parts[1]) {
+		if !labels.IsDNS1123Label(parts[1]) && !gatewaySemantics {
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:2146]: why change this

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 2]
```diff
@@ -981,24 +985,44 @@ func extractParentReferenceInfo(gateways map[parentKey][]*parentInfo, routeRefs
 			SectionName: ptr.OrEmpty(ref.SectionName),
 			Port:        ptr.OrEmpty(ref.Port),
 		}
+		gk := ir
+		if ir.Kind == gvk.Service || ir.Kind == gvk.ServiceEntry {
+			gk = meshParentKey
+		}
+		bannedHostnames := sets.New[string]()
```
Comment (keithmattix) [inline pilot/pkg/config/kube/gateway/conversion.go:992]: What's the point of this variable? seems like it gets shadowed by the other var on L997

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 3]
```diff
@@ -1939,13 +1964,44 @@ type routeParentReference struct {
...
+		// * routeHost, the hostname in the route entry
+		// * ourListener, the hostname of the listener the route is bound to
+		// * checkListener, the hostname of the other listener we are comparing to
+		// We want to return false if checkListener would match the routeHost and it would be a more exact match
+		if len(ourListener) > len(checkListener) {
+			// If our hostname is longer, it must be more exact than the check
+			continue
+		}
+		// Ours is shorter. If it matches the checkListener, then it should ONLY match that one
+		// Note protocol, port, etc are already considered
```
Comment (keithmattix) [inline pilot/pkg/config/kube/gateway/conversion.go:1997]: where are they considered?

File: pilot/pkg/config/kube/gateway/conversion.go (hunk 2 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/conversion.go:992]: hmm, I had a lot of iterations. I think this is deadcode. Nice find

File: pilot/pkg/config/kube/gateway/conversion.go (hunk 3 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/conversion.go:1997]: In appendParent when we compute bannedHostnames. Ill update the comment

**Review by ericvn — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Pull request merged**